sdk/src/calibration_96tof1.cpp

```diff
@@ -205,120 +352,152 @@ aditof::Status Calibration::getIntrinsic(float key,
+    }
+    buildGeometryCalibrationCache(cameraMatrix, frameWidth, frameheight);
 
-    return Status::OK;
+    return status;
 }
 
-//! ReadCalMap - Read the entire calibration map
+//! calibrateDepth - Calibrate the depth data
 /*!
-    ReadCalMap - Read the entire calibration map from a binary file
-    \param calibration_map - Calibarion map to be read
-    \param file_name - File from where the calibration map needs to be read
+calibrateDepth - Calibrate the depth data using the gain and offset
+\param frame - Buffer with the depth data, used to return the calibrated data
+\param frame_size - Number of samples in the frame data
 */
-aditof::Status
-Calibration::readCalMap(std::shared_ptr<aditof::DeviceInterface> device) {
+aditof::Status Calibration96Tof1::calibrateDepth(uint16_t *frame,
+                                                 uint32_t frame_size) {
     using namespace aditof;
 
-    Status status = Status::OK;
-    uint8_t *data;
-    float read_size = 100;
-    uint32_t j = 0;
-    float key;
+    uint16_t *cache = m_depth_cache;
 
-    device->writeEeprom(EEPROM_SIZE - 5, (uint8_t *)&read_size, 4);
+    uint16_t *end = frame + (frame_size - frame_size % 8);
+    uint16_t *framePtr = frame;
 
-    device->readEeprom(0, (uint8_t *)&read_size, 4);
-    LOG(INFO) << "EEPROM calibration data size " << read_size << " bytes";
+    for (; framePtr < end; framePtr += 8) {
+        *framePtr = *(cache + *framePtr);
+        *(framePtr + 1) = *(cache + *(framePtr + 1));
+        *(framePtr + 2) = *(cache + *(framePtr + 2));
+        *(framePtr + 3) = *(cache + *(framePtr + 3));
+        *(framePtr + 4) = *(cache + *(framePtr + 4));
+        *(framePtr + 5) = *(cache + *(framePtr + 5));
+        *(framePtr + 6) = *(cache + *(framePtr + 6));
+        *(framePtr + 7) = *(cache + *(framePtr + 7));
+    }
 
-    if (read_size > EEPROM_SIZE) {
-        LOG(WARNING) << "Invalid calibration data size";
-        return Status::GENERIC_ERROR;
+    end += (frame_size % 8);
+
+    for (; framePtr < end; framePtr++) {
+        *framePtr = *(cache + *framePtr);
     }
 
-    data = (uint8_t *)malloc(read_size);
+    return Status::OK;
+}
 
-    status = device->readEeprom(4, data, (size_t)read_size);
-    if (status != Status::OK) {
-        free(data);
-        LOG(WARNING) << "Failed to read from eeprom";
-        return status;
+//! calibrateCameraGeometry - Compensate for lens distorsion in the depth data
+/*!
+calibrateCameraGeometry - Compensate for lens distorsion in the depth data
+\param frame - Buffer with the depth data, used to return the calibrated data
+\param frame_size - Number of samples in the frame data
+*/
+aditof::Status Calibration96Tof1::calibrateCameraGeometry(uint16_t *frame,
+                                                          uint32_t frame_size) {
+    using namespace aditof;
+
+    for (uint32_t i = 0; i < frame_size; i++) {
+        frame[i] = static_cast<uint16_t>(
```

Comment:
cast to double is not needed. frame[i] will be promoted to double then multiplied with m_geometry_cache[i].

Reply:
Done